Packs/CiscoEmailSecurity/Integrations/CiscoEmailSecurity/CiscoEmailSecurity.py

```diff
@@ -0,0 +1,629 @@
+            json_data=request_body
+        )
+
+    def list_release_quarantine_messages(self, url_suffix, request_body):
+        return self.http_request(
+            method='POST',
+            url_suffix=url_suffix,
+            json_data=request_body
+        )
+
+    def list_entries_get(self, url_suffix):
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix
+        )
+
+    def list_entries_add(self, url_suffix, request_body):
+        return self.http_request(
+            method='POST',
+            url_suffix=url_suffix,
+            json_data=request_body
+        )
+
+    def list_entries_delete(self, url_suffix, request_body):
+        return self.http_request(
+            method='DELETE',
+            url_suffix=url_suffix,
+            json_data=request_body
+        )
+
+
+def test_module(client: Client) -> str:
+    suffix_url = "/sma/api/v2.0/message-tracking/messages?startDate=2018-01-01T00:00:00.000Z&" \
+                 "endDate=2019-11-20T09:36:00.000Z&ciscoHost=All_Hosts&searchOption=messages&offset=0&limit=20"
+    try:
+        client.list_messages(suffix_url)
+    except DemistoException as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+    return 'ok'
+
+
+def date_to_cisco_date(date):
+    return date.replace(' ', 'T') + '.000Z'
+
+
+def set_limit(limit):
+    return int(limit) if limit and int(limit) <= MAX_MESSAGES_TO_GET else MAX_MESSAGES_TO_GET
+
+
+def build_url_params_for_list_report(args):
+    start_date = date_to_cisco_date(args.get('start_date'))
+    end_date = date_to_cisco_date(args.get('end_date'))
+    device_type = args.get('device_type')
+    url_params = f'?startDate={start_date}&endDate={end_date}&device_type={device_type}'
+
+    arguments = assign_params(**args)
+
+    for key, value in arguments.items():
+        if key == 'offset':
+            limit = arguments.get('limit')
+            url_params += f'&{key}={int(value)}&limit={int(limit)}'
+
+        elif key == 'filter_key':
+            filter_operator = arguments.get('filter_operator', 'is')
+            filter_value = arguments.get('filter_value')
+            url_params += f'&filterBy={value}&filter_operator={filter_operator}&filter_value={filter_value}'
+
+        elif key == 'device_group':
+            url_params += f'&{key}={value}'
+        elif key == 'device_name':
+            url_params += f'&{key}={value}'
+
+    return url_params
+
+
+def list_report_command(client: Client, args: Dict[str, Any]):
+    url_suffix = '/sma/api/v2.0/reporting'
+    url_params = build_url_params_for_list_report(args)
+    url_suffix_to_filter_by = url_suffix + url_params
+    report_response_data = client.list_report(url_suffix_to_filter_by)
+    return CommandResults(
+        readable_output='human_readable',
+        outputs_prefix='CiscoEmailSecurity.report',
+        outputs_key_field='',
```

Comment:
there is no key?

Reply:
This command - `cisco-email-security-report-get` doesn't exist in swagger, and in the API docs it appears without any example - so its implementation is not clear.